### PR TITLE
feature/STJS-461-support-apple-pay-visa-checkout-with-cybertonica

### PR DESCRIPTION
**Pull request by pgs-gmartowski**

### CODE REVIEW

File: src/application/core/integrations/VisaCheckout.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ export class VisaCheckout {
 
   protected attachVisaButton = () => DomMethods.appendChildIntoDOM(this._placement, this.createVisaButton());
 
-  protected onSuccess(payment: object) {
+  protected async onSuccess(payment: object) {
```
Comment (wirus15) [inline src/application/core/integrations/VisaCheckout.ts:153]: `async` is not needed here

File: src/application/core/integrations/Cybertonica.ts [hunk 2]
```diff
@@ -26,7 +27,7 @@ export class Cybertonica implements ICybertonica {
     this.tid = DomMethods.insertScript(Cybertonica.SCRIPT_TARGET, { src: Cybertonica.SDK_ADDRESS }).then(
       () => AFCYBERTONICA.init(apiUserName) || this.initFailed()
     );
-
+    this.tid.then(tid => this.storage.setItem(Cybertonica.TID_KEY, tid));
```
Comment (wirus15) [inline src/application/core/integrations/Cybertonica.ts:30]: remember to always add an empty line before `return`

File: src/client/ST.ts [hunk 3]
```diff
@@ -38,6 +38,7 @@ import { BrowserSessionStorage } from '../shared/services/storage/BrowserSession
 import { Notification } from '../application/core/shared/Notification';
 import './../styles/notification.css';
 import { ConfigProvider } from '../application/core/services/ConfigProvider';
+import { Cybertonica } from '../application/core/integrations/Cybertonica';
```
Comment (wirus15) [inline src/client/ST.ts:41]: this is unused?

File: src/application/core/shared/Payment.ts [hunk 4]
```diff
@@ -48,7 +51,7 @@ export class Payment {
       ...merchantData,
       ...payment
     };
-    const cybertonicaTid = await this._cybertonica.getTransactionId();
+    const cybertonicaTid = this._storage.getItem('app.tid');
```
Comment (wirus15) [inline src/application/core/shared/Payment.ts:54]: you have a constant for that key in Cybertonica class, maybe it should be public

Comment (pgs-gmartowski) [reply src/application/core/shared/Payment.ts]: implementation has been changed, I've restored previous function

File: src/client/ST.ts (hunk 3 above)
Comment (pgs-gmartowski) [reply src/client/ST.ts]: yes, deleted

File: src/application/core/integrations/Cybertonica.ts (hunk 2 above)
Comment (pgs-gmartowski) [reply src/application/core/integrations/Cybertonica.ts:30]: done

File: src/application/core/integrations/VisaCheckout.ts (hunk 1 above)
Comment (pgs-gmartowski) [reply src/application/core/integrations/VisaCheckout.ts]: done

**Pull request merged**